aiobotocore/endpoint.py

```diff
@@ -39,7 +39,9 @@ def text_(s, encoding='utf-8', errors='strict'):
 @asyncio.coroutine
 def convert_to_response_dict(http_response, operation_model):
     response_dict = {
-        'headers': http_response.headers,
+        # botocore converts keys to str, so make sure that they are in
+        # the expected case.
+        'headers': {k.lower(): v for k, v in http_response.headers.items()},
```

Comment:
If I don't miss anything, headers are converted to strings here: https://github.com/boto/botocore/blob/76866243555eead44fb8d0cf709ffed2b823a003/botocore/parsers.py#L218

And then `botocore` fails to find the region here: https://github.com/boto/botocore/blob/bb60fedc64bae2c86934067fea563f52db232a73/botocore/utils.py#L940